### DISCUSSION TITLE
How do you manually call transitions without x-show?

### DISCUSSION THREAD

**Discussion by ryatkins** [alpinejs/alpine#886] [1. Help]
Hi All,

**Overview**
I'm making a [popper](https://popper.js.org/) like dropdown component, one that is aware of the screen limits.  Pop the dropdown up if it is going to go off the bottom of the screen and vice-versa (see demo below).

All implementation directions (IntersectionObserver, el.getBoundingClientRect(), etc) of this type of component cannot use `display: none` as you cannot get the Intersection or the size of a hidden component.  Toggling the `visibility` property via `visibility: hidden` does work, but means I cannot use x-show and leverage the native x-transition events.

**Question**
Without using x-show, how can I manually call all the states of my x-transition stack? 
I'd prefer to not have to bind them all manually!  

Access to the transitionIn() and tranistionOut() methods would be ideal here to kick off the stack of transitions.

**Demo**
![ezgif com-gif-maker](https://user-images.githubusercontent.com/1211767/99921670-f68f5700-2cf9-11eb-823c-7e3eaf204857.gif)

**Comment by KevinBatdorf** [comment]
Could you run the positioning logic in `nextTick()` when the element already has display property set?

**Comment by ryatkins** [comment]
@KevinBatdorf - I ended up tweaking the IntersectionObserver code a bit and found that I could get it to detect dropdown most likely when it was coming out of `display: none` and starting the transition.
